### PR TITLE
microsoft graph user gcc high support

### PR DESCRIPTION
**Pull request by YaelShamai**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: (https://jira-dc.paloaltonetworks.com/browse/CIAC-6392)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml [hunk 1]
```diff
@@ -6,12 +6,28 @@ commonfields:
...
+  section: Connect
+  advanced: true
 - defaultvalue: https://graph.microsoft.com
-  display: Host URL (e.g., https://graph.microsoft.com)
+  display: Server URL (e.g., https://graph.microsoft.com)
   name: host
   required: true
   type: 0
   section: Connect
+  additionalinfo: Use this option when required to customize the URL to the Azure management endpoint. More information can be found here - https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#using-national-cloud
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml:30]:
```suggestion
  additionalinfo: Use this option when required to customize the URL to the Microsoft Graph User endpoint. More information can be found here - https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#using-national-cloud
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml [hunk 2]
```diff
@@ -6,12 +6,28 @@ commonfields:
...
+  required: false
+  type: 15
+  options:
+  - Worldwide
+  - US GCC
+  - US GCC-High
+  - DoD
+  - Germany
+  - China
+  additionalinfo: When selecting the Custom option, the Azure AD endpoint parameter must be filled. More information about National clouds can be found here - https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#using-national-cloud
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.yml:21]:
```suggestion
  additionalinfo: When selecting the Custom option, the the Server URL parameter must be filled. More information about National clouds can be found here - https://xsoar.pan.dev/docs/reference/articles/microsoft-integrations---authentication#using-national-cloud
```

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 3]
```diff
@@ -474,7 +487,8 @@ def revoke_user_session_command(client: MsGraphClient, args: dict):
 
 def main():
     params: dict = demisto.params()
-    url = params.get('host', '').rstrip('/') + '/v1.0/'
+    azure_cloud = get_azure_cloud(params, 'MicrosoftGraphUser')
+    url = azure_cloud.endpoints.microsoft_graph_resource_id.rstrip("/") + '/v1.0/'
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py:491]: let's use urljoin
```suggestion
    url = urljoin(azure_cloud.endpoints.microsoft_graph_resource_id, 'v1.0/')
```

File: rce/integration-MicrosoftGraphUser.yml [hunk 4]
```diff
@@ -0,0 +1,2842 @@
+category: Analytics & SIEM
```
Comment (michal-dagan) [inline rce/integration-MicrosoftGraphUser.yml:1]: please revert

File: Packs/MicrosoftGraphUser/ReleaseNotes/1_5_39.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#### Integrations
+
+##### Azure Active Directory Users
+
+- %%UPDATE_RN%%
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphUser/ReleaseNotes/1_5_39.md:5]: seems like a dev-bug

File: Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py [hunk 6]
```diff
@@ -474,7 +487,8 @@ def revoke_user_session_command(client: MsGraphClient, args: dict):
 
 def main():
     params: dict = demisto.params()
-    url = params.get('host', '').rstrip('/') + '/v1.0/'
+    azure_cloud = get_azure_cloud(params, 'MicrosoftGraphUser')
+    url = urljoin(azure_cloud.endpoints.microsoft_graph_resource_id.rstrip("/"), '/v1.0/')
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphUser/Integrations/MicrosoftGraphUser/MicrosoftGraphUser.py:491]:
```suggestion
    url = urljoin(azure_cloud.endpoints.microsoft_graph_resource_id, '/v1.0/')
```
maybe add constant: API_VERSION: str = 'v1.0'
like in MicrosoftGraphDeviceManagement.py

File: Packs/MicrosoftGraphMail/ReleaseNotes/1_6_17.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### Microsoft Graph Mail Single User
+
+- %%UPDATE_RN%%
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphMail/ReleaseNotes/1_6_17.md:6]: same (maybe search all)

File: Packs/MicrosoftCloudAppSecurity/ReleaseNotes/2_2_6.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### Microsoft Defender for Cloud Apps Event Collector
+
+- %%UPDATE_RN%%
```
Comment (michal-dagan) [inline Packs/MicrosoftCloudAppSecurity/ReleaseNotes/2_2_6.md:6]: same

File: Packs/AzureSecurityCenter/ReleaseNotes/2_0_32.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+
+#### Integrations
+
+##### Microsoft Defender for Cloud Event Collector
+
+- %%UPDATE_RN%%
```
Comment (michal-dagan) [inline Packs/AzureSecurityCenter/ReleaseNotes/2_0_32.md:6]: .

File: Packs/MicrosoftGraphUser/ReleaseNotes/1_5_39.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+#### Integrations
+
+##### Azure Active Directory Users
+
+- Added support for All Azure Cloud environments: Public, GCC, GCC-High, DoD, Germany, China.
+- Updated the Docker image to: *demisto/crypto:1.0.0.111961*.
+
+##### Microsoft Graph API
+
+- Added support for All Azure Cloud environments: Public, GCC, GCC-High, DoD, Germany, China.
```
Comment (michal-dagan) [inline Packs/MicrosoftGraphUser/ReleaseNotes/1_5_39.md:10]:
```suggestion
- Added an infrastructure support for Microsoft Graph Application Endpoints.
```

**Review by michal-dagan — APPROVED** [review]
Great!

**Review by JasBeilin — APPROVED** [review]

File: rce/integration-MicrosoftGraphUser.yml [hunk 11]
```diff
@@ -0,0 +1,2841 @@
...
+
+            access_token = response_json.get('access_token', '')
+            expires_in = int(response_json.get('expires_in', 3595))
+
+            return access_token, expires_in, ''
+
+        def _get_self_deployed_token_auth_code(
+                self, refresh_token: str = '', resource: str = '', scope: str | None = None) -> tuple[str, int, str]:
+            """
+            Gets a token by authorizing a self deployed Azure application.
```
Comment (ShirleyDenkberg) [inline rce/integration-MicrosoftGraphUser.yml:1791]:
```suggestion
            Gets a token by authorizing a self-deployed Azure application.
```

File: rce/integration-MicrosoftGraphUser.yml [hunk 12]
```diff
@@ -0,0 +1,2841 @@
...
+                err = f'{str(e)}'
+
+            return_error(f'Error in Microsoft authorization with Azure Managed Identities: {err}')
+            return None
+
+        def _get_token_device_code(
+            self, refresh_token: str = '', scope: str | None = None, integration_context: dict | None = None
+        ) -> tuple[str, int, str]:
+            """
+            Gets a token by authorizing a self deployed Azure application.
```
Comment (ShirleyDenkberg) [inline rce/integration-MicrosoftGraphUser.yml:1871]:
```suggestion
            Gets a token by authorizing a self-deployed Azure application.
```

File: rce/integration-MicrosoftGraphUser.yml [hunk 13]
```diff
@@ -0,0 +1,2841 @@
...
+        """
+           Performs basic GET request to check if the API is reachable and authentication is successful.
+           Returns ok if successful.
+           """
+        response = 'ok'
+        if demisto.params().get('self_deployed', False):
+            if demisto.command() == 'test-module':
+                if client.ms_client.grant_type != CLIENT_CREDENTIALS:
+                    # cannot use test module due to the lack of ability to set refresh token to integration context
+                    # for self deployed app
```
Comment (ShirleyDenkberg) [inline rce/integration-MicrosoftGraphUser.yml:2541]:
```suggestion
                    # for self-deployed app
```

**Comment by ShirleyDenkberg** [comment]
@JasBeilin @michal-dagan Doc review completed.

File: rce/integration-MicrosoftGraphUser.yml [hunk 14]
```diff
@@ -0,0 +1,2841 @@
...
+    name: msgraph-user-session-revoke
+  - description: Generate the login url used for Authorization code flow.
+    name: msgraph-user-generate-login-url
+    arguments: []
+  - description: Run this command if for some reason you need to rerun the authentication process.
+    name: msgraph-user-auth-reset
+    arguments: []
+  dockerimage: demisto/crypto:1.0.0.100307
+  runonce: false
+  script: >
```
[the remark this answers is not in the document]
Comment (YaelShamai) [reply rce/integration-MicrosoftGraphUser.yml:616]: Yes, this is a file to send to the costumer so he can test the integration, it is not a regular yml file.

File: rce/integration-MicrosoftGraphUser.yml [hunk 15]
```diff
@@ -0,0 +1,2841 @@
+sectionOrder:
```
Comment (michal-dagan) [inline rce/integration-MicrosoftGraphUser.yml:1]: please revert

Comment (michal-dagan) [reply rce/integration-MicrosoftGraphUser.yml]: Delete the file from the PR

**Pull request merged**